src/pytest_workflow/file_tests.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) 2018 Leiden University Medical Center
+# This file is part of pytest-workflow
+#
+# pytest-workflow is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# pytest-workflow is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with pytest-workflow.  If not, see <https://www.gnu.org/licenses/
+
+"""All tests for workflow files"""
+import hashlib
+from pathlib import Path
+from typing import Union
+
+import pytest
+
+from .schema import FileTest
+
+
+class FileTestCollector(pytest.Collector):
+    """This collector returns all tests for one particular file"""
+
+    def __init__(self, parent: pytest.Collector, filetest: FileTest,
+                 cwd: Union[bytes, str]):
+        """
+        Create a FiletestCollector
+        :param parent: The collector that started this collector
+        :param filetest: a FileTest object
+        :param cwd: the working directory from which relative filepaths should
+        be evaluated
+        """
+        name = filetest.path.__str__()
+        super().__init__(name, parent)
+        self.filetest = filetest
+        self.cwd = Path(cwd)
+
+    def collect(self):
+        """Returns all tests for one file. Also the absolute path of the files
+        is determined in this function."""
+        filepath = self.filetest.path \
```

Comment:
Bracket method looks a bit cleaner than blackslashes for continuations on multipe lines